dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -830,7 +830,11 @@ private BigDecimal getUnsignedDecimal(int numDigits, int precision, int dataLen)
         BigDecimal ret = new BigDecimal(0);
 
         int idx = 0;
-        int curDigit = numDigits - 1;
+        int curDigit = numDigits;
+        if ((numDigits % 2) == 0)
```

Comment:
It's not entirely obvious to me why this logic is needed, perhaps an inline comment?

Reply:
Good point, added a comment and squashed